pelita/team.py

```diff
@@ -82,10 +82,9 @@ def walls_to_graph(walls, shape=None):
                 # all fields
                 for delta_x, delta_y in [(1, 0), (0, 1)]:
                     neighbor = (x + delta_x, y + delta_y)
-                    # we don't need to check for getting neighbors out of the maze
-                    # because our mazes are all surrounded by walls, i.e. our
-                    # deltas will not put us out of the maze
-                    if neighbor not in walls:
+                    # check if the new node is free and within the bounds as
+                    # the walls might not be closed
+                    if neighbor not in walls and neighbor[0] < width and neighbor[1] < height:
```

Comment:
I guess this could be changed so that the `if` exits faster when one condition is false? Wouldn't this be the same logic?
```python
if neighbor in walls or neighbor[0] >= width or neighbor[1] >= height:
    continue
```
Given that this function is one of the bottlenecks in maze generation but also, unfortunately, also for `Bot`, because the graph gets generated once for each `Bot` instance, it would be cool to optimize as much as possible. I am not sure my suggestion above does indeed improve things, but I guess it should?
```

Reply:
In the game it should only run once per team (and otherwise be cached), so I think it is ok. Also the function spends 60% of the time in `graph.add_edge` which we don’t have any control over. At best, we can therefore half the time.